src/lucky/routeable.cr

```diff
@@ -19,16 +19,39 @@ module Lucky::Routeable
     # **See also** our guides for more information and examples:
     # * [Routing](https://luckyframework.org/guides/actions-and-routing/#routing)
     macro {{ http_method.id }}(path)
-      \{% unless path.starts_with?("/") %}
-        \{% path.raise "Path must start with a slash. Example: '/" + path + "'" %}
-      \{% end %}
-
-      add_route :{{ http_method.id }}, \{{ path }}, \{{ @type.name.id }}
-
-      setup_call_method(\{{ yield }})
+      match(:{{ http_method.id }}, \{{ path }}) do
+        \{{ yield }}
+      end
     end
   {% end %}
 
+  # Define a route with a custom HTTP method.
+  #
+  # Use this method if you need to match a route with a custom HTTP method (verb).
+  # For example:
+  #
+  # ```
+  # class Profile::Show
+  #   match :options, "/profile" do
+  #     # action code here
+  #   end
+  # end
+  #
+  # Will respond to an `HTTP OPTIONS` request.
+  macro match(method, path)
```

Comment:
🤔 I wonder if this should actually take a splat of methods here so it can match on more than one.